contracts/Genesis.sol

```diff
@@ -226,25 +254,85 @@ contract Genesis is ERC721Pausable, VRFConsumerBase, Ownable {
+        external
+        payable
+        whenNotPaused
+        seedGenerated
+        onlyWhitelist(nonce, proof)
+    {
+        require(
+            (tokenCounter.current() + WHITELIST_MINT_COUNT) < MAX_SUPPLY,
+            "Not enough supply"
+        );
         require(msg.value >= PRICE, "Not enough ETH");
-        require((addressToMintCount[msg.sender] + addressToRequestCount[msg.sender]) < WHITELIST_MINT_COUNT, "Already minted");
-        requestMint();
+        require(
+            addressToMintCount[msg.sender] < WHITELIST_MINT_COUNT,
+            "Already minted"
+        );
+        uint256 tokenId = tokenCounter.current();
+        mint(msg.sender, tokenId, generateRandomNumber(tokenId));
     }
 
     /**
-     * Requests a random number to Chainlink VRF for a given token id
-     * @param sender sender of the request 
-     * @param tokenId id of the token
-     * @return requestId id of the request sent to Chainlink
+     * Mint reserved gods
+     * @param to address to send the god to
+     * @param count number of gods to transfer
+     */
+    function mintReservedGods(address to, uint256 count) public onlyOwner {
+        require(
+            reservedGodsSupply() >= count + reservedGodsTransfered.current(),
+            "Not enough reserved gods left"
+        );
+        // Here we don't need to increment counter and god supply counter because we already do in the constructor
+        // to not initialize the counters at 0
+        for (uint256 i = 0; i < count; i++) {
+            reservedGodsTransfered.increment();
+            mintGod(to, reservedGodsTransfered.current());
+        }
+    }
+
+    /**
+     * Will request a random number from Chainlink to be stored privately in the contract
      */
-    function requestRandomNumberForTokenId(address sender, uint256 tokenId) internal returns (bytes32 requestId) {
+    function initializeRandomization() external onlyOwner {
+        require(seed == 0, "Seed already generated");
+        require(randomizationRequestId == 0, "Seed already requested");
         require(LINK.balanceOf(address(this)) >= fee, "Not enough LINK");
-        requestId = requestRandomness(keyHash, fee);
-        requestIdToSender[requestId] = sender;
-        requestIdToTokenId[requestId] = tokenId;
-        return requestId;
+        randomizationRequestId = requestRandomness(keyHash, fee);
+        emit RequestedRandomNumber(randomizationRequestId);
+    }
+
+    /**
+     * Callback when a random number gets generated
+     * @param requestId id of the request sent to Chainlink
+     * @param randomNumber random number returned by Chainlink
+     */
+    function fulfillRandomness(bytes32 requestId, uint256 randomNumber)
+        internal
+        override
+    {
+        console.log("Request Id is");
+        console.logBytes32(requestId);
+        console.log("randomizationRequestId is");
+        console.logBytes32(randomizationRequestId);
```

Comment:
are we leaving this here and stripping them before deployment with the plugin?
if so we need to check out https://github.com/ItsNickBarry/hardhat-log-remover#readme

Reply:
shouldn't be there indeed, will remove, and good plugin, will add!